pyam/_ops.py

```diff
@@ -1,24 +1,39 @@
 import operator
 import pandas as pd
-from pyam.index import append_index_level, get_index_levels
+from pyam.index import append_index_level, get_index_levels, replace_index_values
 from pyam.utils import to_list
+from iam_units import registry
+from pint import Quantity
 
 
 # these functions have to be defined explicitly to allow calling them with keyword args
 def add(a, b):
-    return operator.add(a, b)
+    return operator.add(*_make_series(a, b))
 
 
 def subtract(a, b):
-    return operator.sub(a, b)
+    return operator.sub(*_make_series(a, b))
 
 
 def multiply(a, b):
-    return operator.mul(a, b)
+    return operator.mul(*_make_series(a, b))
 
 
 def divide(a, b):
-    return operator.truediv(a, b)
+    return operator.truediv(*_make_series(a, b))
+
+
+def _make_series(a, b):
+    """Cast instances of a pint.Quantity to a pd.Series
+
+    Calling an operation on a pd.Series and a pint.Quantity removes the units,
+    therefore a pint.Quantity is transformed to a pd.Series of quantities.
+    """
+    if isinstance(a, Quantity):
+        return pd.Series([a] * len(b), index=b.index), b
```

Comment:
We should probably add explicit tests for this.

[Pint-pandas](https://github.com/hgrecco/pint-pandas) is an attempt to support something like this. At the moment I don't think it works for pyam because the entire series has to have the same unit, but maybe something to keep in mind for future.